# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: js/src/jit-test/tests/ion/bug1919246-2.js
# Commit: e88a00447043
# Full Hash: e88a004470438bca174824203a6438444875e71f
# Author: Jan de Mooij <jdemooij@mozilla.com>
# Date: 2024-09-18 04:13:51
# Description:
#   Bug 1919246 - Back out D221953 and add tests. r=mgaudet
#   
#   For now back out D221953 because it was an optional change to improve codegen a bit
#   for loops that contain an OSR loop, but it's probably not worth the complexity to
#   try to handle this.
# ==============================================================================

diff -r a7c3ba245e04 -r e88a00447043 js/src/jit-test/tests/ion/bug1919246-2.js
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/js/src/jit-test/tests/ion/bug1919246-2.js	Tue Sep 17 16:29:02 2024 +0000
@@ -0,0 +1,12 @@
+// |jit-test| --fast-warmup
+function f() {
+  var i = 0;
+  while (true) {
+    for (let a = 2; a;) {
+      for (c = 0; c < 10;) {
+        if (i++ > 2500) return;
+      }
+    }
+  }
+}
+f();